Makefile

```diff
@@ -77,16 +85,42 @@ test:
 doc:
 	$(CARGO) doc
 
+e2e: build
+	(cd ./api_tests; yarn install; yarn test)
+
 check_format: check_rust_format check_toml_format check_ts_format
 
+STAGED_FILES = $(shell git diff --staged --name-only)
+STAGED_RUST_FILES = $(filter %.rs,$(STAGED_FILES))
```

Comment:
Is that only executed if the invoked task is below?

Reply:
No, I think that is always executed. Why?

Reply:
I was hoping for it to be optimized.

Reply:
It is pretty freakin' fast :)